### PR TITLE
checker: add pkgconfig to comptime if

### PR DESCRIPTION
**Pull request by LouisSchmieder**
This pr add the possibility to check if a pkgconfig exists via `$if pkgconfig.{pkgname} {`. This is needed for mysql package, because newer system only have mariadb, but both api's are basicly the same

### CODE REVIEW

**Comment by spytheman** [comment]
Please add test cases.

**Comment by spytheman** [comment]
`a.c` should not be part of the PR

File: vlib/v/gen/c/comptime.v [hunk 1]
```diff
@@ -381,6 +382,14 @@ fn (mut g Gen) comp_if_cond(cond ast.Expr) bool {
...
 			return true
 		}
+		ast.SelectorExpr {
+			left := cond.expr as ast.Ident
+			pkg := cond.field_name
+			if left.name == 'pkgconfig' {
+				g.write('${check_pkgconfig(pkg)}')
+			}
+			return true
+		}
```
Comment (spytheman) [inline vlib/v/gen/c/comptime.v:392]: why is that needed?
the checker would have already run pkgconfig, and determined whether the condition is true or false

**Comment by spytheman** [comment]
I think that `$if pkgconfig.xyz {` is a mistake.
There are pkgconfig names like `gio-unix-2.0` .

**Comment by spytheman** [comment]
How about `$if $pkgconfig('gio-unix-2.0') {` instead ?

File: vlib/v/gen/c/comptime.v (hunk 1 above)
Comment (LouisSchmieder) [reply vlib/v/gen/c/comptime.v:392]: because otherwise it would always generate ```c
#if 1
{
}
#else
{
//code here
}
```
and we should not edit the ast in the checker afaik

**Comment by LouisSchmieder** [comment]
yeah can change to that

**Comment by radare** [comment]
Very nice idea!

**Comment by LouisSchmieder** [comment]
thanks

File: vlib/v/gen/c/comptime.v (hunk 1 above)
Comment (spytheman) [reply vlib/v/gen/c/comptime.v:387]: the checker already sets various things, like types and caches comptime calculations
the more it does, the less complex cgen (and the other backends) can be
